### PR TITLE
Add note on experimental semantic convention implementation

### PR DESCRIPTION
**Pull request by lalitb**
Fixes 968

## Changes

- Add comment for semantic convention for resource and trace - to indicate it's based on experimental specs, and may change in future.
- Rename experimental headers with `experimenta_*` prefix
- Rename few src files having `.cpp` extension to `.cc` to be consistent with other src files.
- 
cc @jsuereth 
For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by ThomsonTan** [comment]
@jsuereth do you think this declaration for semantic convention is sufficient, or we also need to declare the status in the release document?

**Comment by lalitb** [comment]
> @jsuereth do you think this declaration for semantic convention is sufficient, or we also need to declare the status in the release document?

We should be adding it in release description for 1.0.0 as done by other SIGs : 
Eg Java https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.5.0

File: api/include/opentelemetry/trace/semantic_conventions.h [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+// NOTE:
```
Comment (jsuereth) [inline api/include/opentelemetry/trace/semantic_conventions.h:4]: Can you also add a note in the "versioning" document that denotes that headers considered unstable will be documented as such?

Comment (lalitb) [reply api/include/opentelemetry/trace/semantic_conventions.h:4]: This is done now.

File: Versioning.md [hunk 2]
```diff
@@ -65,6 +65,12 @@ Refer to the [ABI Policy](./docs/abi-policy.md) for more details. To summarise:
   allowed to break existing stable interfaces. Feature flags will be removed
   once we have a stable implementation for the signal.
 
+* As an exception, small experimental features in otherwise stable signals/components
+  mayn't necessarily be released under feature flag. These would be flagged as experimental
+  by adding a `NOTE` in it's header file - either at the beginning of file, or as the comment for
```
Comment (jsuereth) [inline Versioning.md:70]: non-blocking comment -

Would it make sense to call the header file `experimental_*` to further clarify?

I think you could do this and still provide backwards compatibility later by forward-chain including the new non-experimental folder when it stabilizes.   Just a thought.

**Review by jsuereth — APPROVED** [review]

File: Versioning.md (hunk 2 above)
Comment (lalitb) [reply Versioning.md:70]: Yes this does makes sense. Will incorporate this.

Comment (lalitb) [reply Versioning.md:70]: This is done now.

**Pull request merged**